src/core/lib/pick-layers.js

```diff
@@ -268,9 +268,11 @@ function getClosestFromPickingBuffer(gl, {
     pickedObjectIndex: -1
   };
 
+  const pickableLayers = layers.filter(layer => layer.isPickable);
```

Comment:
By the way, does this cover all picking cases? This file is really confusing with functions calling functions calling draw.... I.e. do all roads end here, going through this if?

Reply:
I moved `isPickable()` check into `getPickedColors` which is shared by all picking flows, good catch.